src/app/content/highlights/components/CardWrapper.tsx

```diff
@@ -66,23 +67,42 @@ const Wrapper = ({highlights, className, container, highlighter}: WrapperProps)
   }, [cardsHeights]);
 
   const getOffsetsForHighlight = React.useCallback((highlight: Highlight) => {
-    if (offsets.has(highlight.id)) {
-      return assertDefined(offsets.get(highlight.id), 'this has to be defined');
-    } else {
-      const newOffsets = assertDefined(
-        getHighlightOffset(container, highlight),
-        `Couldn't get offsets for highlight with an id: ${highlight.id}`
-      );
-      setOffsets((state) => new Map(state).set(highlight.id, newOffsets));
-      return newOffsets;
-    }
-  }, [container, offsets]);
+    const newOffsets = assertDefined(
+      getHighlightOffset(container, highlight),
+      `Couldn't get offsets for highlight with an id: ${highlight.id}`
+    );
+    setOffsets((state) => new Map(state).set(highlight.id, newOffsets));
+    return newOffsets;
+  }, [container]);
+
+  const checkIfHiddenByCollapsedAncestor = (highlight: Highlight) => {
+    const highlightElement = highlight.elements[0] as HTMLElement;
+    const collapsedAncestor = highlightElement
+      ? highlightElement.closest('[data-type="solution"][aria-expanded="false"]')
```

Comment:
you could check in general for visibility, instead of specifically for collapsed solutions. there are a couple ways of doing that

Reply:
I could recursively look for a parent with overflow: hidden and check if
```js
highlightSpan.getBoundingClientRect().bottom > containerWithOverflowHidden.getBoundingClientRect().bottom
```
but it seems kinda clanky.

Reply:
the problem is that we're styling the solution containers with `overflow: hidden; height: 0`, i'm not sure why we're doing that, if you use either `display: none` or `visibility: hidden` then `node.offsetParent` will be null when it isn't visible

Reply:
@TomWoodward 
I've found the explanation:
```js
// hide text visually, but make rangy still search through it
// when highlighting a search result
// https://github.com/timdown/rangy/wiki/Text-Range-Module#visible-text
const initialSolutionStyles = 'display: block; overflow: hidden; height: 0px';
```

Reply:
ugh... if we changed the solution expander to use details/summary block, would highlights still work then? this solution wouldn't require javascript and has been proposed to be moved upstream to content generation. at first glance i don't see display block or visibility hidden on the collapsed details elements

Reply:
seems like checking for general visibility should work now that you're using the details element

Reply:
i'm not even totally sure rangy will search in collapsed details elements, so thats still an open question